include/fiction/utils/math_utils.hpp

```diff
@@ -49,6 +50,79 @@ T integral_abs(const T n) noexcept
 
     return static_cast<T>(std::abs(static_cast<int64_t>(n)));  // needed to solve ambiguity of std::abs
 }
+/**
+ * Calculates the binomial coefficient C(n, k).
+ *
+ * @param n The total number of items.
+ * @param k The number of items to choose from n.
+ * @return The binomial coefficient C(n, k).
+ */
+inline uint64_t binomial_coefficient(uint64_t n, uint64_t k) noexcept
```

Comment:
I like your implementation a lot! The dynamic programming aspect of it is great. You managed to use O(1) space complexity and O(n) runtime complexity.

Someone made a cool implementation here: https://github.com/BiagioFesta/algorithms/blob/master/src/DynamicProgramming/BinomialCoefficient.cpp Maybe you want to check it out.

Reply:
@marcelwa Thanks a lot! When implementing the binomial coefficient, there are many ways to do it wrong. I got inspired by Wikipedia.  

I will check it out. Thanks!